src/path.cc

```diff
@@ -268,4 +267,42 @@ std::string PathResolve(Environment* env,
 }
 #endif  // _WIN32
 
+#ifdef _WIN32
+constexpr static bool IsWindowsDeviceRoot(const char code) noexcept {
+  return (code >= 'a' && code <= 'z') || (code >= 'A' && code <= 'Z');
+}
+#endif  // _WIN32
+
+void ToNamespacedPath(Environment* env, BufferValue* path) {
+#ifdef _WIN32
+  if (path->length() == 0) return;
+  auto resolved_path = node::PathResolve(env, {path->ToStringView()});
+
+  if (resolved_path.size() <= 2) {
+    path->SetLength(resolved_path.size());
+    memcpy(&path, resolved_path.c_str(), resolved_path.size());
+    return;
+  }
+
+  if (resolved_path[0] == '\\') {
+    // Possible UNC root
+    if (resolved_path[1] == '\\') {
+      if (resolved_path[2] != '?' && resolved_path[2] != '.') {
+        // Matched non-long UNC root, convert the path to a long UNC path
+        std::string intermediary_path =
+            "\\\\?\\UNC\\" + resolved_path.substr(2);
+        path->SetLength(intermediary_path.size());
+        memcpy(&path, intermediary_path.c_str(), intermediary_path.size());
```

Comment:
I am concerned that you might be allocating an `std::string` instance unnecessarily. Maybe something like this instead:

```C++
std::string_view unc_prefix = R"(\\?\UNC\")";
std::string_view resolved_path2 = resolved_path.substr(2);
path->SetLength(unc_prefix.size() + resolved_path2());
memcpy(&path, unc_prefix.data(), unc_prefix.size());
memcpy(&path +  unc_prefix.size(), resolved_path2.data(), resolved_path2());
```

Reply:
I love it. Nice catch. Thank you @lemire